metricbeat/module/couchbase/bucket/data.go

```diff
@@ -0,0 +1,69 @@
+package bucket
+
+import (
+	"encoding/json"
+	"io"
+
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/logp"
+)
+
+type BucketQuota struct {
+	RAM    int64 `json:"ram"`
+	RawRAM int64 `json:"rawRAM"`
+}
+
+type BucketBasicStats struct {
+	QuotaPercentUsed float64 `json:"quotaPercentUsed"`
+	OpsPerSec        int64   `json:"opsPerSec"`
+	DiskFetches      int64   `json:"diskFetches"`
+	ItemCount        int64   `json:"itemCount"`
+	DiskUsed         int64   `json:"diskUsed"`
+	DataUsed         int64   `json:"dataUsed"`
+	MemUsed          int64   `json:"memUsed"`
+}
+
+type Buckets []struct {
+	Name       string           `json:"name"`
+	BucketType string           `json:"bucketType"`
+	Quota      BucketQuota      `json:"quota"`
+	BasicStats BucketBasicStats `json:"basicStats"`
+}
+
+func eventsMapping(body io.Reader) []common.MapStr {
+
+	var d Buckets
+	err := json.NewDecoder(body).Decode(&d)
+	if err != nil {
+		logp.Err("Error: ", err)
+	}
+
+	events := []common.MapStr{}
+
+	for _, Bucket := range d {
+		event := common.MapStr{
+			"name": Bucket.Name,
+			"type": Bucket.BucketType,
+			"data": common.MapStr{
+				"used.bytes": Bucket.BasicStats.DataUsed,
+			},
+			"disk": common.MapStr{
+				"fetches":    Bucket.BasicStats.DiskFetches,
+				"used.bytes": Bucket.BasicStats.DiskUsed,
```

Comment:
Unfortuantely for compatibility reasons also the `.bytes` and `.pct` have to be inside their own common.MapStr. I hope we can simplify this in the future similar to what we did on the fields.yml side.